Ttv 121 course create for merge

pekMant opened the pull request:
This is a proposal to merge a new feature to TIM-JYU/TIDE-CLI: a create-course functionality as a subcommand for the task command group.

The purpose of create-course function is to expand the task retrieval capabilities of the TIDE-CLI program. The original create function has two options for task retrieval and creation: create single task by task id or create all tasks using document path. The new functionality uses TIM API endpoint tasksByCourse to retrieve all tasks included in a one course. This retrieval can be done using either document path or course id.

The response TIM generates from a request by course-create is handled by new get_tasks_by_course() function in routes.py. The response sent by TIM contains a list of tasksets, which are handled so that correct folder structure is generated for the course tasks.

The commits also include preliminary unit tests to ensure the correct operation of the course-create function. The tests are currently fairly large, as they are made from the example course data from integration tests. If requested these tests can be reworked to more manageable size and expanded in tested functionalities.

The course-create function is planned to be used in VSCode TIDE extension, as an option when dowloading tasksets.

sjsarsa commented:
On a quick glance, looks good. Perhaps it would be more in line with existing commands if this was  under a subcommand `course` and the command name `task`, i.e. `tide course create` as in the existing `tide task create`.

pekMant commented:
I created a new command group "course" and added "create" (previously create-course) under it. 

Now that there is a command group for course related commands, it could be good to move "courses" command under it. The courses command lists all courses the user has access to with tide-cli. Courses command is called only once in Vscode extension tide commands, so in our project it would be easy to change. This would require communication with the other group working with tide-cli.

Additionally, I added some missing help texts to options and updated the unit tests to the changed command name.

sjsarsa commented:
Thank you,

and agreed, `courses` command could be `course list` now. A new PR would be welcome.
